test/manual/templates/system_api_test.sh

```
@@ -0,0 +1,648 @@
+# Enhanced response parsing for more robust handling
+parse_response_safe() {
+  local raw="$1"
+  
+  # If empty input, return empty JSON
+  if [[ -z "$raw" ]]; then
+    echo "{}"
+    return
+  fi
+  
+  # Extract the JSON part from a HTTP response (ignore headers)
+  local json_part=$(echo "$raw" | grep -v "^HTTP/" | grep -v "^[[:space:]]*$" | tail -n 1)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)